SegmentMesher/SegmentMesher.py

```diff
@@ -50,211 +50,122 @@ def setup(self):
-
-    #
-    # Display
-    #
-    displayParametersCollapsibleButton = ctk.ctkCollapsibleButton()
-    displayParametersCollapsibleButton.text = "Display"
-    displayParametersCollapsibleButton.collapsed = True
-    self.layout.addWidget(displayParametersCollapsibleButton)
-    displayParametersFormLayout = qt.QFormLayout(displayParametersCollapsibleButton)
-
-    self.clipNodeWidget=slicer.qMRMLClipNodeWidget()
+    self.ui.customTetGenPathSelector.setCurrentPath(customTetGenPath)
+    self.ui.customTetGenPathSelector.nameFilters = [self.logic.tetGenFilename]
+   
     clipNode = slicer.mrmlScene.GetFirstNodeByClass("vtkMRMLClipModelsNode")
-    self.clipNodeWidget.setMRMLClipNode(clipNode)
-    displayParametersFormLayout.addRow(self.clipNodeWidget)
-
-    #
-    # Apply Button
-    #
-    self.applyButton = qt.QPushButton("Apply")
-    self.applyButton.toolTip = "Run the algorithm."
-    self.applyButton.enabled = False
-    self.layout.addWidget(self.applyButton)
-
-    self.statusLabel = qt.QPlainTextEdit()
-    self.statusLabel.setTextInteractionFlags(qt.Qt.TextSelectableByMouse)
-    self.statusLabel.setCenterOnScroll(True)
-    self.layout.addWidget(self.statusLabel)
+    self.ui.clipNodeWidget.setMRMLClipNode(clipNode)
+    
 
     # connections
-    self.applyButton.connect('clicked(bool)', self.onApplyButton)
-    self.showTemporaryFilesFolderButton.connect('clicked(bool)', self.onShowTemporaryFilesFolder)
-    self.inputModelSelector.connect("currentNodeChanged(vtkMRMLNode*)", self.updateMRMLFromGUI)
-    self.outputModelSelector.connect("currentNodeChanged(vtkMRMLNode*)", self.updateMRMLFromGUI)
-    self.methodSelectorComboBox.connect("currentIndexChanged(int)", self.updateMRMLFromGUI)
+    self.ui.applyButton.connect('clicked(bool)', self.onApplyButton)
+    self.ui.showTemporaryFilesFolderButton.connect('clicked(bool)', self.onShowTemporaryFilesFolder)
+    self.ui.inputModelSelector.connect("currentNodeChanged(vtkMRMLNode*)", self.updateMRMLFromGUI)
+    self.ui.inputSurfaceSelector.connect("currentNodeChanged(vtkMRMLNode*)", self.updateMRMLFromGUI)
+    self.ui.outputModelSelector.connect("currentNodeChanged(vtkMRMLNode*)", self.updateMRMLFromGUI)
+    self.ui.methodSelectorComboBox.connect("currentIndexChanged(int)", self.updateMRMLFromGUI)
     # Immediately update deleteTemporaryFiles in the logic to make it possible to decide to
     # keep the temporary file while the model generation is running
-    self.keepTemporaryFilesCheckBox.connect("toggled(bool)", self.onKeepTemporaryFilesToggled)
+    self.ui.keepTemporaryFilesCheckBox.connect("toggled(bool)", self.onKeepTemporaryFilesToggled)
+    self.ui.tetgenUseSurface.connect("toggled(bool)", self.updateMRMLFromGUI)
 
     # Add vertical spacer
     self.layout.addStretch(1)
 
     # Refresh Apply button state
     self.updateMRMLFromGUI()
 
+  def enter(self):
+    self.updateMRMLFromGUI()
+  
   def cleanup(self):
     pass
 
+   
   def updateMRMLFromGUI(self):
```

Comment:
It would be nice to save widgets state into a parameter node (see how it is done in the module template)